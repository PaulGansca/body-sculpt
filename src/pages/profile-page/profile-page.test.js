import { shallow } from 'enzyme';
import React from 'react';
import { Provider } from 'react-redux';
import ProfilePage from './profile-page';
import ProfilePageContainer from './profile-page-container';

let wrapper;

beforeEach(() => {
    const mockProps = {
        currentUser: {id: "bla123bla", fitnessLevel: "intermediate"},
        loading: false,
        workouts: [{date: new Date()}],
    }
    wrapper = shallow(<ProfilePage {...mockProps} />);
})

describe('<ProfilePage />', () => {
    it("expect to render without crashing with no props", () => {
        const mockProps = {
            currentUser: "",
            userLoading: true,
            getState: jest.fn(),
            subscribe: jest.fn(),
            dispatch: jest.fn(),
            workouts: [{date: new Date()}],
        }
        wrapper = shallow(
                    <Provider store={mockProps}>
                        <ProfilePageContainer {...mockProps} />
                    </Provider>)
        expect(wrapper).toMatchSnapshot();
    })
    it("expect to render profile page with props", () => {
        expect(wrapper).toMatchSnapshot();
    })
});