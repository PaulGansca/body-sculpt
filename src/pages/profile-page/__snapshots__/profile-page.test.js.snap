// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProfilePage /> expect to render profile page with props 1`] = `ShallowWrapper {}`;

exports[`<ProfilePage /> expect to render without crashing with no props 1`] = `ShallowWrapper {}`;
